panther_lights/src/controller_node.py

```diff
@@ -21,10 +20,77 @@
 
 
 class PantherAnimation:
+    ANIMATION_DEFAULT_PRIORITY = 3
+    ANIMATION_DEFAULT_TIMEOUT = 120
+
     front: Animation
     rear: Animation
-    interrupting = False
-    repeating = False
+    name: str = 'UNDEFINED'
+    priority: int = ANIMATION_DEFAULT_PRIORITY
+    timeout: float = ANIMATION_DEFAULT_TIMEOUT
+    repeating: bool = False
+
+    def __init__(self) -> None:
+        self._init_time = rospy.get_time()
+
+    def reset_time(self) -> None:
+        self._init_time = rospy.get_time()
+
+    @property
+    def init_time(self) -> float:
+        return self._init_time
+
+
+class AnimationsQueue:
+    def __init__(self, max_queue_size) -> None:
+        self._queue = []
+        self._max_queue_size = max_queue_size
+
+    def put(self, animation: PantherAnimation) -> None:
+        if animation.priority == 1:
+            self.clear()
+        self.validate_queue()
+
+        if len(self._queue) == self._max_queue_size:
+            rospy.logwarn(f'{rospy.get_name()} Animation queue overloaded')
+            return
+
+        self._queue.append(animation)
+        self._queue = sorted(self._queue, key=lambda x: (x.priority, x.init_time))
+
+    def get(self) -> PantherAnimation:
+        return self._queue.pop(0)
+
+    def empty(self) -> bool:
+        return len(self._queue) == 0
+
+    def clear(self, priority: int = 2) -> None:
+        # clears queue except items below specified priority
+        self._queue = [animation for animation in self._queue if animation.priority < priority]
+
+    def remove(self, animation) -> None:
+        if self.has_animation(animation):
+            self._queue.remove(animation)
+
+    def get_first_anim_priority(self) -> int:
```

Comment:
```suggestion
    def first_anim_priority(self) -> int:
```
Since this is only function with `get` prefix I think it can be skipped. Also this could be an property